data-prepper-plugins/saas-source-plugins/saas-crawler/src/main/java/org/opensearch/dataprepper/plugins/source/saas/crawler/base/Crawler.java

```diff
@@ -0,0 +1,85 @@
+package org.opensearch.dataprepper.plugins.source.saas.crawler.base;
+
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.EnhancedSourceCoordinator;
+import org.opensearch.dataprepper.plugins.source.saas.crawler.coordination.partition.SaasSourcePartition;
+import org.opensearch.dataprepper.plugins.source.saas.crawler.coordination.state.SaasWorkerProgressState;
+import org.opensearch.dataprepper.plugins.source.saas.crawler.model.ItemInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Named;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Named
+public class Crawler {
+    private static final Logger log = LoggerFactory.getLogger(Crawler.class);
+    private static final int maxItemsPerPage = 20;
+    public static final String CREATED = "created";
+    public static final String UPDATED = "updated";
+
+    private final SaasClient client;
+
+    public Crawler(SaasClient client) {
+        this.client = client;
+    }
+
+    public long crawl(long lastPollTime,
+                      EnhancedSourceCoordinator coordinator) {
+        long startTime = System.currentTimeMillis();
+        client.setLastPollTime(lastPollTime);
+        Iterator<ItemInfo> itemInfoIterator = client.listItems();
+        log.info("Starting to crawl the source");
+        long updatedPollTime = 0;
+        log.info("Creating Partitions");
```

Comment:
There are a log of logs throughout the PR that are `info` logs that look like they should be `debug`

Reply:
Adjusted the log level for some and removed a few log statements.